File: yframework-ddd/src/main/java/org/yframework/ddd/common/domain/IRepository.java
package org.yframework.ddd.common.domain;

import java.util.List;

/**
 * Description: 仓储<br>
 * Comments Name: IRepository<br>
 * Date: 2019/12/3 09:23<br>
 * Author: ysj<br>
 * EditorDate: 2019/12/3 09:23<br>
 * Editor: ysj
 */
public interface IRepository<T extends IDomainObject>
{
    List<T> list(T t);

    int size(T t);

    T get(T t);

    int add(T t);

    int update(T t);

    int remove(T t);
}

File: yframework-android/framework/src/main/java/org/yframework/android/common/IRepository.kt
package org.yframework.android.common


/**
 * Description: IRepository<br>
 * Comments Name: IRepository<br>
 * Date: 2019-08-23 10:56<br>
 * Author: ysj<br>
 * EditorDate: 2019-08-23 10:56<br>
 * Editor: ysj
 */
interface IRepository<T> {
    fun list(t: T?): List<T>?

    fun size(t: T?): Int

    fun get(t: T): T?

    fun add(t: T): Int

    fun update(t: T): Int

    fun remove(t: T): Int
}